### PR TITLE
Automation logging updates

### PR DESCRIPTION
**Pull request by Rory-Powell**
## Description
Updates to logging for better pino integration to aid debugging via datadog. 

Additionally add some better type definitions around bull jobs specific to automations.

### CODE REVIEW

File: packages/backend-core/src/logging/pino/logger.ts
Comment (Rory-Powell) [reply packages/backend-core/src/logging/pino/logger.ts:169]: The reason is that this may be invoked anywhere a console statement is done. We may or may not be in automation context at this point, or in any context at all. The safest approach is to try-catch so that we don't break logging.

File: packages/backend-core/src/queue/listeners.ts
Comment (Rory-Powell) [inline packages/backend-core/src/queue/listeners.ts:114]: I'm planning on turning off `bull` debug logging by default to bring down logging costs, hence the `stalled` and `error` logs above being promoted to logging by default

**Review by mike12345567 — APPROVED** [review]
LGTM! Would be good after getting this merged to pull it back to development as some of the types changes will have implications for sync automations - theres a few changes around `Job` types already!

File: packages/backend-core/src/context/mainContext.ts
Comment (adrinr) [inline packages/backend-core/src/context/mainContext.ts:108]: Can we use an object as a parameter to make it more verbose in its usage?
```
export async function doInAutomationContext(params: {
  appId: string,
  automationId: string,
  task: any
})
```

**Review by adrinr — COMMENTED** [review]
Small comment, but LGTM

**Comment by PClmnt** [comment]
> LGTM! Would be good after getting this merged to pull it back to development as some of the types changes will have implications for sync automations - theres a few changes around `Job` types already!

+1 to this, some of the stuff you've done around the triggers and automation thread code overlap quite heavily with some work I was doing for sync automations. LGTM though!

**Pull request merged**
